### PR TITLE
DOP-3267

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

[DOP-3267](https://jira.mongodb.org/browse/DOP-3267) Add embedded Version Select in ToC

[design prototypes](https://www.figma.com/proto/WGKRPJVpCjmvuNJxFcy7xn/Embedded-Version-Support?page-id=120%3A3092&node-id=120%3A3961&viewport=918%2C776%2C0.09&scaling=contain&starting-point-node-id=120%3A3961)

### Staging Links:

[atlas docs with embedded version testing data](https://docs-mongodbcom-integration.corp.mongodb.com/ce842592fbf45d611a1d5214a476d021188e0bc8/master/cloud-docs/seung.park/DOP-3267/)
[atlas cli docs with merged ToC](https://docs-mongodbcom-integration.corp.mongodb.com/ce842592fbf45d611a1d5214a476d021188e0bc8/master/atlas-cli/seung.park/DOP-3267/install-atlas-cli/) 
**note** the links for atlas pages are not correctly spoofed. also will get 404's when navigating to different versions as they require separate deployments. therefore, will _correctly_ be only able to see the most recent version as selected
[server docs for regression check on select](https://docs-mongodbcom-integration.corp.mongodb.com/7ea6a7d33f7973e9eda82c90f6020b2a4e69f250/master/docs/seung.park/DOP-3267/tutorial/insert-documents/)

### Notes:
- if you want to check out the spoofed data, using test data from snooty_dev.metadata with `project: "atlas-cli"` and `project:"cloud-docs"`. these have spoofed merged `toctree` `project` and `associated_fields` properties

### CODE REVIEW

File: src/components/Sidenav/TOCNode.js [hunk 1]
```diff
@@ -30,6 +53,7 @@ const TOCNode = ({ activeSection, handleClick, level = BASE_NODE_LEVEL, node })
   const { title, slug, url, children, options = {} } = node;
   const target = slug || url;
   const hasChildren = !!children.length;
+  const hasVersions = !!(options?.versions?.length > 1); // in the event there is only one version, do we show version selector?
```
Comment (casthewiz) [inline src/components/Sidenav/TOCNode.js:56]: If there's only one version, no need to show the version selector afaik 👍

File: src/components/Sidenav/VersionSelector.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    // via difference of snooty.toml[associated_products] vs pool.repo_branches
+    setOptions(buildChoices(availableVersions[versionedProject]));
+  }, [availableVersions, versionedProject]);
+
+  const navigateToVersion = (gitBranchName) => {
+    const branch = availableVersions[versionedProject].find((branch) => branch['gitBranchName' === gitBranchName]);
+    const slug = branch?.slug;
+    if (slug) {
+      console.log('navigating to slug ', slug);
+    }
```
Comment (casthewiz) [inline src/components/Sidenav/VersionSelector.js:54]: Do we want to remove the console.log here before merge?

**Review by casthewiz — COMMENTED** [review]
Staging link looks really good in general - it's so exciting to see this working live!

I think we'll need to set the default selection on a user first visiting the page (seems like active version behavior is correct, it's just not displaying `stable` in the dropdown when visiting the page for the first time in incognito)

File: src/components/Sidenav/VersionSelector.js (hunk 2 above)
Comment (seungpark) [reply src/components/Sidenav/VersionSelector.js:54]: this actually requires some elements from [DOP-3266](https://github.com/mongodb/snooty/pull/710/files#diff-0e4db20c7678504170e62381114917943edbd8739de8c5442b32acf9535fec01) as they share logic in parsing from git branch to target url slug

**Comment by seungpark** [comment]
> Staging link looks really good in general - it's so exciting to see this working live!
> 
> I think we'll need to set the default selection on a user first visiting the page (seems like active version behavior is correct, it's just not displaying `stable` in the dropdown when visiting the page for the first time in incognito) edit: not sure if this is what you meant by "ToC does not refresh on initial load. may have to refresh the page"

it's a bug with setting the default version if not found. this has been addressed with https://github.com/mongodb/snooty/pull/716

discussed offline and merged that logic into this pr

**Comment by seungpark** [comment]
@casthewiz i made a feature branch for the two issues DOP-3267 and DOP-3266 since they rely on each other. can merge to main branch after verifying both features + unit tests

**Comment by sandynguyenn** [comment]
For [atlas docs with embedded version testing data](https://docs-mongodbcom-integration.corp.mongodb.com/35a40e54609f1a5d7f4da7564184d68385cfe7a6/master/cloud-docs/seung.park/DOP-3267/)

- In the design prototype, the selector input is only supposed to show on hover but its in active state in staging. Is this one of the constraints you mentioned @seungpark ? With the current active selector state in the TOC, I worry that it will be distracting to users when scanning other category titles -- hence why the original design has the selector in an inactive state that is less distracting.

**Comment by seungpark** [comment]
> For [atlas docs with embedded version testing data](https://docs-mongodbcom-integration.corp.mongodb.com/35a40e54609f1a5d7f4da7564184d68385cfe7a6/master/cloud-docs/seung.park/DOP-3267/)
> 
> * In the design prototype, the selector input is only supposed to show on hover but its in active state in staging. Is this one of the constraints you mentioned @seungpark ? With the current active selector state in the TOC, I worry that it will be distracting to users when scanning other category titles -- hence why the original design has the selector in an inactive state that is less distracting.

good catch @sandynguyenn ! last commit addresses this and i updated the staging links above

**Review by sandynguyenn — APPROVED** [review]
lgtm

**Review by allisonmui9 — CHANGES_REQUESTED** [review]
Thanks Seung for moving so quick! Just a few comments: 

- Sorry for the inconvenience, but can we change the button to what you had before? where it showed the button right away? I think it looks more clean, but having the button and its hover state blend in with the rest of the menu (especially when being hovered on) take away from the usability. I think having it show the button style is ultimately better for the user's usability. 
- When the user clicks the button and the menu expands, can we have the caret in the button change to caret up? 

Thank you!

**Comment by seungpark** [comment]
> Thanks Seung for moving so quick! Just a few comments:
> 
> * Sorry for the inconvenience, but can we change the button to what you had before? where it showed the button right away? I think it looks more clean, but having the button and its hover state blend in with the rest of the menu (especially when being hovered on) take away from the usability. I think having it show the button style is ultimately better for the user's usability.
> * When the user clicks the button and the menu expands, can we have the caret in the button change to caret up?
> 
> Thank you!

@allisonmui9  sounds good! thank you for the catch on accessibility 👍 
as for the caret, this is default behavior from LG/select components https://www.mongodb.design/component/select/example/ i'll open a request ticket for this caret feature if that works with you

**Review by allisonmui9 — APPROVED** [review]
lgtm thank you!

**Review by rayangler — APPROVED** [review]
This looks really cool! Just one question about whether we should remove some commented code

**Pull request merged**